vernac/vernacentries.mli

```diff
@@ -27,3 +27,36 @@ val interp_redexp_hook : (Environ.env -> Evd.evar_map -> Genredexpr.raw_red_expr
 val command_focus : unit Proof.focus_kind
 
 val allow_sprop_opt_name : string list
+
+(** pre-processing and validation of VernacInductive *)
+module Preprocessed_Mind_decl : sig
+  type flags = {
+    template : bool option;
+    udecl : Constrexpr.cumul_univ_decl_expr option;
+    cumulative : bool;
+    poly : bool;
+    finite : Declarations.recursivity_kind;
+  }
+  type record = {
+    flags : flags;
+    primitive_proj : bool;
+    kind : Vernacexpr.inductive_kind;
+    records : Record.Ast.t list;
+  }
+  type inductive = {
+    flags : flags;
+    typing_flags : Declarations.typing_flags option;
+    private_ind : bool;
+    uniform : ComInductive.uniform_inductive_flag;
+    inductives : ((Names.lident * Vernacexpr.inductive_params_expr * Constrexpr.constr_expr option * Vernacexpr.constructor_expr list) * Vernacexpr.decl_notation list) list;
```

Comment:
Don't we have an alias for this long type?

Reply:
done